operator/internal/status/lokistack.go

```diff
@@ -149,53 +156,19 @@ func checkForZoneawareNodes(ctx context.Context, k client.Client, zones []lokiv1
 	return true, true, nil
 }
 
-func updateCondition(ctx context.Context, k k8s.Client, req ctrl.Request, condition metav1.Condition) error {
-	var stack lokiv1.LokiStack
-	if err := k.Get(ctx, req.NamespacedName, &stack); err != nil {
-		if apierrors.IsNotFound(err) {
-			return nil
-		}
-		return kverrors.Wrap(err, "failed to lookup LokiStack", "name", req.NamespacedName)
+func generateWarnings(schemas []lokiv1.ObjectStorageSchema) []metav1.Condition {
+	warnings := make([]metav1.Condition, 0)
+	if len(schemas) == 0 {
+		return nil
```

Comment:
The return value of `generateWarnings` is used to `append` the other conditions to, so I let it always return an initialized slice, even though it might be empty (GH does not let me make a suggestion for the whole function, so I can only list the new function):

```go
func generateWarnings(schemas []lokiv1.ObjectStorageSchema) []metav1.Condition {
	warnings := make([]metav1.Condition, 0, 2)

	if len(schemas) > 0 && schemas[len(schemas)-1].Version != lokiv1.ObjectStorageSchemaV13 {
		warnings = append(warnings, metav1.Condition{
			Type:    string(lokiv1.ConditionWarning),
			Reason:  string(lokiv1.ReasonStorageNeedsSchemaUpdate),
			Message: messageWarningNeedsSchemaVersionUpdate,
		})
	}

	return warnings
}
```